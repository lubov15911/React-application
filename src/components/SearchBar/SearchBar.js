import React from 'react';
import PropTypes from 'prop-types';

import './SearchBar.scss';

export default class SearchBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    onSubmit(event) {
        const { handleSubmit } = this.props;
        const { value } = this.state;

        handleSubmit(value);
        event.preventDefault();
    }

    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    render() {
        const { value } = this.state;
        return (
            <form className="search-form" onSubmit={this.onSubmit}>
                <input type="text" placeholder="Search" value={value} onChange={this.handleChange} />
                <button type="submit"><p>Search</p></button>
            </form>
        );
    }
};
SearchBar.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
};
