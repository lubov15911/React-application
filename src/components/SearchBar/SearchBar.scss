.search-form {
  display: grid;
  grid-template-columns: 2fr 1fr;

  input,
  button {
    height: 2.8em;
    border-radius: 4px;
    border: none;
    color: inherit;
    box-sizing: border-box;
    font-size: 2em;
  }

  input {
    opacity: 0.7;
    background-color: rgba(50, 50, 50, 0.8);
    margin-right: 1em;
    padding: 0 0.7em;
  }

  button {
    background-color: #f65261;
    cursor: pointer;
  }
}
