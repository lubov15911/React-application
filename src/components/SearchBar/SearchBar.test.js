import React from 'react';
import { shallow } from 'enzyme';

import SearchBar from './SearchBar';

describe('SearchBar', () => {
    const fakeFunctionSubmit = jest.fn();
    const fakeFunctionSearchValue= jest.fn();
    const searchValue = 'Tarantino';
    let changeEvent;
    let component;

    beforeEach(() => {
        changeEvent = { target: { value: searchValue } };
        component = shallow(<SearchBar
            searchValue={searchValue}
            handleSubmit={fakeFunctionSubmit}
            handleSearchValue={fakeFunctionSearchValue} />);
    });

    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should change value state', () => {
        component.find('input').simulate('change', changeEvent);
        expect(fakeFunctionSearchValue).toHaveBeenCalledWith(searchValue);
    });

    it('should submit current search value', () => {
        const submitEvent = { preventDefault: () => {} };
        component.find('input').simulate('change', changeEvent);
        component.find('form').simulate('submit', submitEvent);
        expect(fakeFunctionSubmit).toHaveBeenCalledWith(searchValue);
    });
});
