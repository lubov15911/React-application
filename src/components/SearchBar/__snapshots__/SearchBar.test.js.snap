// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="Tarantino"
  />
  <button
    type="submit"
  >
    <p>
      Search
    </p>
  </button>
</form>
`;
